### ISSUE TITLE
[Input Date Picker] Start/End dates don't adjust initially when animations are reduced/disabled

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#6800]
### Actual Behavior

Input Date Picker start/end dates don't adjust initially when:
- Windows animations are disabled (Chrome, FF, Edge browsers), and
- Mac's animations are reduced (Chrome, FF browsers)

![input-date-picker-animation-bug](https://user-images.githubusercontent.com/5023024/232538266-c537d951-7a4d-402c-a8a3-9315605c15c6.gif)

cc Created from an internal request (151) @jlias-devtester and #6410

### Expected Behavior

When users have reduced motion, they should have similar experiences with `input-date-picker` changed values on initialization.

### Reproduction Sample

https://codepen.io/geospatialem/pen/XWPOPaq

### Reproduction Steps

1. Disable animations (Windows) or enable reduced animations (Mac)
2. [Open the sample](https://codepen.io/geospatialem/pen/XWPOPaq) in either:
    - Windows: Chrome, FF, Edge, or
    - Mac: Chrome, FF 
4. Open the `modal` by selecting the "Open modal" button
5. Try to modify the `date-picker`'s start or end date.
6. Observe the initial setting does not alter
7. Try to modify the start or end date again.
8. Observe the change works.
9. Enable animations (Windows) or disable reduced animations (Mac)
10. Repeat steps 2-6, and observe all changes are reflected as-expected

### Reproduction Version

1.1.0

### Relevant Info

- Windows - all browsers (Chrome, FF, Edge)
- Mac - Chrome and FF browsers (does not impact Safari)

### Regression?

1.0.8

### Priority impact

p4 - not time sensitive

### Impact

Seems relatively low, but looks to be a regression - works as-expected in `1.0.8`, and is presented in `1.1.0`.

Possibly related to https://github.com/Esri/calcite-components/commit/dfcaa228708fc9d57c74a1c41cd6a9033874d026, if the animation fix includes `input-date-picker`? cc @jcfranco 

### Esri team

ArcGIS Online

**Comment by jcfranco** [comment]
@geospatialem This is no longer reproducible with 1.3.0+.

**Comment by geospatialem** [comment]
Thanks @jcfranco, am also unable to repo in newer releases. Closing!
